releasenotes/notes/add-filter-98d00f306b5689ee.yaml

```
@@ -0,0 +1,20 @@
+---
+features:
+  - |
+    The :class:`~rustworkx.PyGraph` and the :class:`~rustworkx.PyDiGraph` classes have a new method 
+    :meth:`~rustworkx.PyGraph.filter_nodes` (or :meth:`~rustworkx.PyDiGraph.filter_nodes`). 
+    This method returns a NodeIndices object with the resulting nodes that fit some abstract criteria indicated by a filter function.
+    For example::
+      def my_filter_function(node)
+        return isinstance(node, str)
+
+      string_indices = graph.filter_nodes(my_filter_function)
```

Consider changing these lines as they're causing `tox -edocs` to fail:
```suggestion
    For example:

    .. jupyter-execute::
    
      from rustworkx import PyGraph

      graph = PyGraph()
      graph.add_nodes_from(list(range(5))) # Adds nodes from 0 to 5

      def my_filter_function(node):
        return isinstance(node, str)

      string_indices = graph.filter_nodes(my_filter_function)
```